src/components/Steps.js

```diff
@@ -175,6 +182,12 @@ export default class Steps extends React.Component {
    * if `opts.ignoreValidation` is true.
    */
   goToStep = async (step, opts = {}) => {
+    const currentStep = this.getStepByName(this.props.currentStepName);
+    const stepPayment = this.props.stepPayment || {};
+    if (currentStep.isLastStep && stepPayment.isNew && (stepPayment.error || !stepPayment.data)) {
```

Comment:
Same as before, would be great to add code comments to explain in plain english what we're doing here.